### ISSUE TITLE
Adding an option that will allow to move all dependencies to the lockfile only

### ISSUE THREAD

**Issue by zkochan** [pnpm/pnpm#6750]
In order to reduce conflicts, we could store dependencies fields in the lockfile only. This would be an opt-in feature for now. On publish, pnpm would add the dependencies to the `package.json` file before publish.

**Comment by await-ovo** [comment]
Is this to avoid lockfile conflicts, or something else?  The lockfile now has dependencies information:

```lockfileVersion: '6.0'

settings:
  autoInstallPeers: true
  excludeLinksFromLockfile: false

dependencies:
  vue:
    specifier: ^3.3.4
    version: 3.3.4

packages:
```

I may not understand the problem trying to solve,  can you explain it to me? In addition, I am happy to implement this, after I understand, thanks.

**Comment by zkochan** [comment]
yes. Although it will make it impossible to change the dependencies by manually changing the ranges in `package.json`. Or by manually removing dependencies.
